lib/ecto/repo/schema.ex

```diff
@@ -706,7 +706,7 @@ defmodule Ecto.Repo.Schema do
   defp dump_field!(action, schema, field, type, value, adapter) do
     case Ecto.Type.adapter_dump(adapter, type, value) do
       {:ok, value} ->
-        {field, value}
+        {schema.__schema__(:source, field), value}
```

Comment:
This happens at the point we no longer use changesets, so I don't see a reason to store it in the changeset.